pkg/apis/resource/extension.go

```diff
@@ -271,17 +267,24 @@ func (h Handler) RouteGetAccessEndpoints(req RouteGetAccessEndpointsRequest) (Ro
 
 func (h Handler) getAccessEndpoints(ctx context.Context, id object.ID) ([]AccessEndpoint, error) {
 	// Endpoints from output.
-	eps, err := h.getEndpointsFromOutput(ctx, id)
+	oeps, err := h.getEndpointsFromOutput(ctx, id)
 	if err != nil {
 		return nil, err
 	}
 
-	if len(eps) != 0 {
-		return eps, nil
+	// Endpoints from resources.
+	reps, err := h.getEndpointsFromResources(ctx, id)
+	if err != nil {
+		return nil, err
 	}
 
-	// Endpoints from resources.
-	return h.getEndpointsFromResources(ctx, id)
+	eps := oeps
+	eps = append(eps, reps...)
```

Comment:
eps := append(oeps, reps...)

Reply:
Because Go append manipulates underlying array, so the `eps := append(oeps, reps...)` will failed with lint below.
`appendAssign: append result not assigned to the same slice (gocritic)`